WSLGd/ProcessMonitor.cpp

```diff
@@ -26,8 +26,21 @@ int wslgd::ProcessMonitor::LaunchProcess(
     if (childPid == 0) try {
         // Construct a null-terminated argument array.
         std::vector<const char*> arguments;
-        for (auto &arg : argv) {
-            arguments.push_back(arg.c_str());
+        if (argv[0].compare("/init") != 0) {
+            for (auto &arg : argv) {
+                arguments.push_back(arg.c_str());
+            }
+        } else {
+            for (int i = 0; i < argv.size(); i++) {
+                arguments.push_back(argv[i].c_str());
+                if (i == 1) {
+                    /* insert executable name without full path after 
+                       full path executable at arguments[2] */
+                    auto const pos = argv[i].find_last_of('/');

```

Comment:
nit: This can be simplified by: `arguments.push_back(basename(argv[i]));`

Reply:
(Also has the benefit of handling the case where '/' isn't in `argv[i] `

Reply:
Changed to use basename instead of hard coded executable name.